astropy/io/fits/header.py

```diff
@@ -1537,9 +1527,27 @@ def remove(self, keyword):
         keyword : str
             The keyword of which to remove the first instance in the header
 
+	ignore_missing : bool, optional
+	    When True, ignores missing keywords.  Otherwise, if the keyword is not present 
+            in the header a `KeyError` is raised.
+
+	all : bool, optional
+	    When True, all instances of keyword will be removed.
+            Otherwise only the first instance of the given keyword is removed.
+
         """
+        	
+        try:
+            del self[self.index(keyword)]
+            if all:
+                while keyword in self._keyword_indices:
+                    del self[self.index(keyword)]		
+        except ValueError:
+            if ignore_missing:
+                pass
+            else:
+                raise KeyError('Keyword %s not in the header.' % keyword)		
 
-        del self[self.index(keyword)]
```

Comment:
Ah, I see now that the original code went through `self.index` too, so that's probably why you kept it that way.  :)  Yeah, it would be a slight enhancement for this to go directly through `self._keyword_indices` instead.

Though it occurs to me now that since the `keyword` variable here is user-input, it should also be "normalized" before being compared with the entries in `self._keyword_indices` (this already happens in self.index which may be why I used it originally).

So before doing anything in this method run:

```
keyword = Card.normalize_keyword(keyword)
```

Users are sometimes inconsistent about how they write FITS keywords--so this puts them in a "normalized" form where they're all upper-case, etc.